app/controllers/posts_controller.rb

```diff
@@ -9,4 +9,30 @@ def show
     @user = @post.user
     @comments = @post.comments
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @new_post = current_user.posts.new(params.require(:post).permit(:title, :text))
+    @new_post.likes_counter = 0
+    @new_post.comments_counter = 0
+    @new_post.update_posts_counter
+    respond_to do |format|
+      format.html do
+        if @new_post.save
+          redirect_to "/users/#{@new_post.user.id}/posts/", notice: 'Successfully saved !'
+        else
+          render :new, alert: 'Error occured!'
+        end
+      end
+    end
+  end
+
+  # private
+
+  # def post_params
+  #   params.require(:data).permit(:title, :text)
+  # end
```

Comment:
- [ ] I would recommend including the private method, it makes the application secure by permitting only limited attributes.

Reply:
Thank you for pointing out that error. Fixed this.